db/upgrade.php

```diff
@@ -72,5 +72,55 @@ function xmldb_tool_advancedreplace_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2024100400, 'tool', 'advancedreplace');
     }
 
+    if ($oldversion < 2024101600) {
+
+        // Define table tool_advancedreplace_search to be created.
+        $table = new xmldb_table('tool_advancedreplace_files');
+
+        // Adding fields to table tool_advancedreplace_search.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $table->add_field('userid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('name', XMLDB_TYPE_CHAR, '32', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('pattern', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('components', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('skipcomponents', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('mimetypes', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('skipmimetypes', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('filenames', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('skipfilenames', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        $table->add_field('skipareas', XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
+        // These next few columns are the same as the search table above.
+        // They have the 8th parameter to avoid cut-and-paste detection in ci phpcpd.
```

Comment:
I think there should be some way to exclude this from phpcpd without modifying the content? Not a big concern, just the current approach looks a bit messy with the inconsistent formatting.

Reply:
I could not find a way to turn off phpcpd.  
And I tried a few white-space changes, but that did not satisfy phpcpd. It seems to still recognise copy-paste even after /*comments*/ and new lines and indentation are added. 

I think this is the best compromise for this situation.  These lines of code only need to run once more - on UAT.  Then nobody will care about them again!